### ISSUE TITLE
Potentially important min violation warnings are lost

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#12732]
Inspired by #12731, and possibly related to #10477. Consider the following MWE:

```modelica
package MinErrorOrWarning
  model Test1 "This should generate a warning about delta being lower than it min attribute"
    Real P;
    parameter Real delta(min=Modelica.Constants.eps) = compute_delta();
    Real q;
    
    function compute_delta
      output Real y;
    algorithm
      y := -3.5;
    end compute_delta;
  
  equation
    P = 0;
    q = if noEvent(P > delta) or noEvent(P < -delta) then 1/P else 0;
  end Test1;
  
  model Test2 "A warning is obtained, not clear how lambda = 0 is managed with the division by zero"
    Real P;
    parameter Real Pmin(min=Modelica.Constants.eps) = compute_Pmin();
    parameter Real delta = Pmin/1000;
    Real q;
    
    function compute_Pmin
      output Real y;
    algorithm
      y := -3.5;
    end compute_Pmin;
    
  equation
    P = homotopy(1,0);
    q = if noEvent(P > delta) or noEvent(P < -delta) then 1/P else 0;
  end Test2;
  
  model Test3 "A warning is obtained, not clear how lambda = 0 is managed with the division by zero"
    Real P;
    parameter Real Pmin(min=Modelica.Constants.eps) = compute_Pmin();
    parameter Real delta = Pmin/1000;
    Real q;
    
    function compute_Pmin
      output Real y;
    algorithm
      y := -3.5;
    end compute_Pmin;
  
  equation
    assert(Pmin > Modelica.Constants.eps, "Pmin should be positive");
    P = homotopy(1,0);
    q = if noEvent(P > delta) or noEvent(P < -delta) then 1/P else 0;
  end Test3;
  
  model Test4 "If one keeps the assert only, then the model actually fails on the assert (why not just a warning?)"
    Real P;
    parameter Real Pmin = compute_Pmin();
    parameter Real delta = Pmin/1000;
    Real q;
    
    function compute_Pmin
      output Real y;
    algorithm
      y := -3.5;
    end compute_Pmin;
  
  equation
    assert(Pmin > Modelica.Constants.eps, "Pmin should be positive");
    P = homotopy(1,0);
    q = if noEvent(P > delta) or noEvent(P < -delta) then 1/P else 0;
  end Test4;
  
  model Test5
    Real P;
    parameter Real Pmin(min=Modelica.Constants.eps) = compute_Pmin();
    parameter Real delta = Pmin/1000;
    Real q;
    
    function compute_Pmin
      output Real y;
    algorithm
      y := -3.5;
    end compute_Pmin;
  initial equation
    assert(Pmin > Modelica.Constants.eps, "Pmin should be positive");
  equation
    P = homotopy(1,0);
    q = if noEvent(P > delta) or noEvent(P < -delta) then 1/P else 0;
  end Test5;
  
  model Test6
    Real P;
    parameter Real Pmin(min=Modelica.Constants.eps) = compute_Pmin();
    parameter Real delta = Pmin/1000;
    Real q;
    
    function compute_Pmin
      output Real y;
    algorithm
      y := -3.5;
    end compute_Pmin;
    
  equation
    P = homotopy(0,0);
    q = if noEvent(P > delta) or noEvent(P < -delta) then 1/P else 0;
  end Test6;
end MinErrorOrWarning;
```
If you run Test1, it fails because of division by zero. The root cause is Pmin being negative (thus violating the min attribute), but there is no warning about that. Some warning should be issued in that case.

In Test2, the homotopy operator was added, and this causes a warning to be triggered at lambda = 0, which is OK. On the other hand, it is not clear how the solver manages to continue, since the problem at lambda = 0 is clearly singular.

Test3, Test4, and Test5all have an assertion, and that makes the model fail (as expected).

Test6 is always singular because of homotopy(0, 0), however no warning on the min attribute violation is triggered.

Bottom line: min violations should always be reported (as warnings, see discussion in #10477), not only in some cases.
